src/main/java/org/javarosa/xform/util/SecondaryInstanceAnalyzer.java

```diff
@@ -0,0 +1,41 @@
+package org.javarosa.xform.util;
+
+import org.kxml2.kdom.Element;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import static org.javarosa.xform.util.XFormSerializer.elementToString;
+
+
+public class SecondaryInstanceAnalyzer {
+    // a list of SIs that need to be built in memory
+    private final List<String> inMemorySecondaryInstances;
+
+    private final Pattern INSTANCE_FUNCTION_PATTERN = Pattern.compile("instance\\s*\\(\\s*'([^\\s]{1,64})'\\s*");
```

Comment:
This regular expression should have some corresponding tests to document its intended behavior.  A comment would help too (white space is allowed after the function name, before the argument, etc). In particular, there should be an explanation for the 64 character limit.

Also note that double quotes are allowed for strings as well as single quotes.